projects/js-packages/publicize-components/src/social-store/actions/connection-data.js

```diff
@@ -202,3 +229,54 @@ export function deleteConnectionById( { connectionId, showSuccessNotice = true }
 		}
 	};
 }
+
+/**
+ * Creates a connection.
+ *
+ * @param {Record<string, any>} data - The data for API call.
+ * @returns {void}
+ */
+export function createConnection( data ) {
+	return async function ( { dispatch } ) {
+		const { createErrorNotice, createSuccessNotice } = coreDispatch( globalNoticesStore );
+
+		try {
+			const path = `/jetpack/v4/social/connections/`;
+
+			dispatch( creatingConnection() );
+
+			const connection = await apiFetch( { method: 'POST', path, data } );
+
+			if ( connection ) {
+				dispatch(
+					addConnection( {
+						...connection,
+						// TODO fix this messy data structure
```

Comment:
This comment was suggested by Copilot. 😄 

Even Copilot knows this should be fixed.